### PR TITLE
[Issue-1429]: Grid Content component in Carnation

### PR DESCRIPTION
**Pull request by afi13**
Issue: https://github.com/ymcatwincities/openy/issues/1429
## Steps for review

- [ ] Enable Carnation theme.
- [ ] Create new (or edit existing) Landing Page.
- [ ] Adde new 'Grid Content' paragraph.
- [ ] Make sure 'Style' setting is used on the front.
- [ ] Make sure there no any visible regression on the front for 2, 3 and 4 items per row. For desktop and mobile version.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [ ] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Can one of the admins verify this patch? Use "o+k to test" or ''t+est this please" for manual build execution.

**Review by ddrozdik — APPROVED** [review]

File: themes/openy_themes/openy_carnation/templates/paragraph/paragraph--grid-content--default.html.twig [hunk 1]
```diff
@@ -45,12 +45,20 @@ set classes = [
...
+{% elseif grid_style == '4' %}
+  {% set item_class = 'col-12 col-sm-3 row-eq-height' %}
+{% endif %}
 <div{{ attributes.addClass(classes) }}>
   {% for key, item in content.field_grid_columns %}
     {% if key matches '/^\\d+$/' %}
-    <div class="col-12 col-lg">
-      {{ item }}
-    </div>
+      <div class="{{ item_class }}">
```
Comment (podarok) [inline themes/openy_themes/openy_carnation/templates/paragraph/paragraph--grid-content--default.html.twig:59]: what if grid_style not in (2,3,4) ?
```class=""``` breaks css validation rules

**Review by podarok — CHANGES_REQUESTED** [review]

File: themes/openy_themes/openy_carnation/templates/paragraph/paragraph--grid-content--default.html.twig (hunk 1 above)
Comment (ddrozdik) [reply themes/openy_themes/openy_carnation/templates/paragraph/paragraph--grid-content--default.html.twig:59]: hmm, at this moment we have only these options. But I agree, let's add a default value.

**Comment by afi13** [comment]
Continue in https://github.com/ymcatwincities/openy/pull/1442

**Comment by afi13** [comment]
Ready for review https://github.com/ymcatwincities/openy/pull/1442

**Pull request closed without merging**